paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/ManageScreenUI.kt

```diff
@@ -68,45 +74,70 @@ internal fun ManageScreenUI(interactor: ManageScreenInteractor) {
 }
 
 @Composable
-private fun TrailingContent(isSelected: Boolean, isEditing: Boolean, isModifiable: Boolean, paymentMethodId: String?) {
+private fun TrailingContent(
+    isSelected: Boolean,
+    isEditing: Boolean,
+    isModifiable: Boolean,
+    paymentMethod: DisplayableSavedPaymentMethod,
+    interactor: ManageScreenInteractor,
+) {
     if (isEditing && isModifiable) {
         Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
-            EditIcon(paymentMethodId)
-            DeleteIcon(paymentMethodId)
+            EditIcon(paymentMethod)
+            DeleteIcon(paymentMethod, interactor)
         }
     } else if (isEditing) {
-        DeleteIcon(paymentMethodId)
+        DeleteIcon(paymentMethod, interactor)
     } else if (isSelected) {
         SelectedBadge()
     }
 }
 
 @Composable
-private fun DeleteIcon(paymentMethodId: String?) {
+private fun DeleteIcon(paymentMethod: DisplayableSavedPaymentMethod, interactor: ManageScreenInteractor) {
+    val openRemoveDialog = rememberSaveable { mutableStateOf(false) }
+    val paymentMethodId = paymentMethod.paymentMethod.id
+
     TrailingIcon(
         backgroundColor = Color.Red,
         icon = Icons.Filled.Close,
-        modifier = Modifier.testTag("${TEST_TAG_MANAGE_SCREEN_DELETE_ICON}_$paymentMethodId")
+        modifier = Modifier.testTag("${TEST_TAG_MANAGE_SCREEN_DELETE_ICON}_$paymentMethodId"),
+        onClick = {
+            openRemoveDialog.value = true
+        },
     )
+
+    if (openRemoveDialog.value) {
+        RemovePaymentMethodDialogUI(paymentMethod = paymentMethod, onConfirmListener = {
+            openRemoveDialog.value = false
+            interactor.handleViewAction(ManageScreenInteractor.ViewAction.DeletePaymentMethod(paymentMethod))
```

Comment:
I think this matches behavior of what we do with horizontal mode. But this means we delete the payment method without showing any loading indicator, or caring if the operation succeeds, right?

Reply:
Yep. Not sure if we have plans to improve this or not